syntax = "proto3";

import "Responses.proto";
import "Requests.proto";

service ModelControllerService {
  rpc AddTask(AddTaskRequest) returns (AddTaskResponse) {}
  rpc AddSubTask(AddSubTaskRequest) returns (AddSubTaskResponse) {}
  rpc EditTask(EditTaskRequest) returns (EditTaskResponse) {}
  rpc DeleteTask(DeleteTaskRequest) returns (DeleteTaskResponse) {}
  rpc CompleteTask(CompleteTaskRequest) returns (CompleteTaskResponse) {}
  rpc AddTaskLabel(AddTaskLabelRequest) returns (AddTaskLabelResponse) {}
  rpc DeleteTaskWithSubTasks(DeleteTaskWithSubTasksRequest) returns (DeleteTaskWithSubTasksResponse) {}
  rpc CompleteTaskWithSubTasks(CompleteTaskWithSubTasksRequest) returns (CompleteTaskWithSubTasksResponse) {}
  rpc GetAllTasks(GetAllTasksRequest) returns (GetAllTasksResponse) {}
  rpc GetTask(GetTaskRequest) returns (GetTaskResponse) {}
  rpc SaveToFile(SaveToFileRequest) returns (SaveToFileResponse) {}
  rpc LoadFromFile(LoadFromFileRequest) returns (LoadFromFileResponse) {}
}